impeller/renderer/backend/vulkan/blit_command_vk.cc

```diff
@@ -144,6 +144,65 @@ bool BlitCopyTextureToBufferCommandVK::Encode(CommandEncoderVK& encoder) const {
   return true;
 }
 
+//------------------------------------------------------------------------------
+/// BlitCopyBufferToTextureCommandVK
+///
+
+BlitCopyBufferToTextureCommandVK::~BlitCopyBufferToTextureCommandVK() = default;
+
+std::string BlitCopyBufferToTextureCommandVK::GetLabel() const {
+  return label;
+}
+
+bool BlitCopyBufferToTextureCommandVK::Encode(CommandEncoderVK& encoder) const {
+  const auto& cmd_buffer = encoder.GetCommandBuffer();
+
+  // cast destination to TextureVK
+  const auto& dst = TextureVK::Cast(*destination);
+  const auto& src = DeviceBufferVK::Cast(*source.buffer);
+
+  if (!encoder.Track(source.buffer) || !encoder.Track(destination)) {
+    return false;
+  }
+
+  LayoutTransition transition;
+  transition.cmd_buffer = cmd_buffer;
+  transition.new_layout = vk::ImageLayout::eTransferSrcOptimal;
+  transition.src_access = vk::AccessFlagBits::eShaderWrite |
+                          vk::AccessFlagBits::eTransferWrite |
+                          vk::AccessFlagBits::eColorAttachmentWrite;
+  transition.src_stage = vk::PipelineStageFlagBits::eFragmentShader |
+                         vk::PipelineStageFlagBits::eTransfer |
+                         vk::PipelineStageFlagBits::eColorAttachmentOutput;
+  transition.dst_access = vk::AccessFlagBits::eShaderRead;
```

Comment:
The mnemonic I use for transitions is "For this resource, all the commands before this may continue to src_access in src_stage and no commands after this proceed past dst_access in dst_stage.". I believe the src accesses here are right. You are waiting for any commands before this blit that write, transfer, specify this resource as color attachment as to finish. But, you are going to be using this as a blit source (I believe these are "transfers" in Vulkan parlance). So commands that read or transfer **from** it must wait.

Reply:
Oh, poop. Sorry, its the other way round. What I said was TextureToBuffer.